### PR TITLE
[storage] update the backupstorage interface

### PR DESCRIPTION
**Pull request by areshand**
### Description
Return file handle in save_xxx interfaces and update the commands
### Test Plan
use existing tests

### CODE REVIEW

File: docker/compose/data-restore/gcs.yaml [hunk 1]
```diff
@@ -11,6 +11,6 @@ commands:
     exec 1>&-  # close stdout
     gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" > /dev/null
   open_for_read: 'gsutil -q cp "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" - | gzip -cd'
-  save_metadata_line: 'gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/metadata/$FILE_NAME"'
+  save_metadata_line: 'gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE"'
```
Comment (msmouse) [inline docker/compose/data-restore/gcs.yaml:14]: sorry but let's do it the hard way like what `create_for_write` does 😂

**Review by grao1991 — APPROVED** [review]

File: storage/backup/backup-cli/src/storage/command_adapter/sample_configs/gcp.sample.yaml [hunk 2]
```diff
@@ -20,8 +20,7 @@ commands:
     # route file handle content to stdout
     gsutil -q cp "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" - | gzip -cd
   save_metadata_line: |
-    # save the line to a new file under the metadata folder
-    gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/metadata/$FILE_NAME"
+    FILE_HANDLE="metadata/$FILE_NAME"; echo "$FILE_HANDLE"; exec 1>&-; gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" > /dev/null
```
Comment (msmouse) [inline storage/backup/backup-cli/src/storage/command_adapter/sample_configs/gcp.sample.yaml:23]: match the style of `create_for_write` please

File: storage/backup/backup-cli/src/storage/command_adapter/sample_configs/s3.sample.yaml [hunk 3]
```diff
@@ -21,7 +21,7 @@ commands:
     aws s3 cp "s3://$BUCKET/$SUB_DIR/$FILE_HANDLE" - | gzip -cd
   save_metadata_line: |
     # save the line to a new file under the metadata folder
-    gzip -c | aws s3 cp - "s3://$BUCKET/$SUB_DIR/metadata/$FILE_NAME"
+    FILE_HANDLE="metadata/$FILE_NAME"; echo "$FILE_HANDLE"; exec 1>&-; gzip -c | aws s3 cp - "s3://$BUCKET/$SUB_DIR/$FILE_HANDLE"
```
Comment (msmouse) [inline storage/backup/backup-cli/src/storage/command_adapter/sample_configs/s3.sample.yaml:24]: ditto

File: storage/backup/backup-cli/src/storage/mod.rs [hunk 4]
```diff
@@ -160,7 +160,11 @@ pub trait BackupStorage: Send + Sync {
     /// Behavior on duplicated names is undefined, overwriting the content upon an existing name
     /// is straightforward and acceptable.
     /// See `list_metadata_files`.
-    async fn save_metadata_line(&self, name: &ShellSafeName, content: &TextLine) -> Result<()>;
+    async fn save_metadata_line(
```
Comment (msmouse) [inline storage/backup/backup-cli/src/storage/mod.rs:163]: update the comment to indicate a filename is returned

File: storage/backup/backup-cli/src/storage/mod.rs [hunk 5]
```diff
@@ -171,7 +175,11 @@ pub trait BackupStorage: Send + Sync {
     /// Move a metadata file to the metadata file backup folder.
     async fn backup_metadata_file(&self, file_handle: &FileHandleRef) -> Result<()>;
     /// Save a vector of metadata lines to file. If the file exists, this will overwrite
-    async fn save_metadata_lines(&self, name: &ShellSafeName, lines: &[TextLine]) -> Result<()>;
+    async fn save_metadata_lines(
```
Comment (msmouse) [inline storage/backup/backup-cli/src/storage/mod.rs:178]: update the comment

File: terraform/helm/fullnode/files/backup/oci.yaml [hunk 6]
```diff
@@ -7,7 +7,7 @@ commands:
     exec 1>&-  # close stdout
     gzip -c | curl -s -X PUT --data-binary "@-" "$ENDPOINT$ACCESS_URI/$SUB_DIR/$FILE_HANDLE"
   open_for_read: 'curl -s "$ENDPOINT$ACCESS_URI/$SUB_DIR/$FILE_HANDLE" | gzip -cd'
-  save_metadata_line: 'gzip -c | curl -s -X PUT --data-binary "@-" "$ENDPOINT$ACCESS_URI/$SUB_DIR/metadata/$FILE_NAME"'
+  save_metadata_line: 'FILE_HANDLE="metadata/$FILE_NAME"; echo "$FILE_HANDLE"; exec 1>&-; gzip -c | curl -s -X PUT --data-binary "@-" "$ENDPOINT$ACCESS_URI/$SUB_DIR/$FILE_HANDLE"'
```
Comment (msmouse) [inline terraform/helm/fullnode/files/backup/oci.yaml:10]: match the style, and all other places, please..

File: terraform/helm/fullnode/files/backup/gcs.yaml [hunk 7]
```diff
@@ -7,6 +7,6 @@ commands:
     exec 1>&-  # close stdout
     gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" > /dev/null
   open_for_read: 'gsutil -q cp "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" - | gzip -cd'
-  save_metadata_line: 'gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/metadata/$FILE_NAME"'
+  save_metadata_line: 'FILE_HANDLE="metadata/$FILE_NAME"; echo "$FILE_HANDLE"; exec 1>&-; gzip -c | gsutil -q cp - "gs://$BUCKET/$SUB_DIR/$FILE_HANDLE" > /dev/null'
```
Comment (msmouse) [inline terraform/helm/fullnode/files/backup/gcs.yaml:10]: match the style

**Review by msmouse — APPROVED** [review]

**Pull request merged**